tests/unit/syncModel/webSpotlightTransformer.test.ts

```diff
@@ -0,0 +1,55 @@
+import { describe, expect, it } from "@jest/globals";
+import { ContentTypeContracts } from "@kontent-ai/management-sdk";
+
+import { transformWebSpotlightModel } from "../../../src/modules/sync/modelTransfomers/webSpotlight";
+
+const contentTypes = [
+  {
+    id: "typeId1",
+    name: "type 1",
+    codename: "type_codename_1",
+    elements: [],
+    last_modified: "",
+  },
+] as const satisfies ReadonlyArray<ContentTypeContracts.IContentTypeContract>;
+
+const createEnvironmentModel = (isWebSpotlightEnabled: boolean, rootTypeId: string | null) => ({
+  contentTypes,
+  contentTypeSnippets: [],
+  taxonomyGroups: [],
+  webSpotlight: {
+    enabled: isWebSpotlightEnabled,
+    root_type: rootTypeId ? { id: rootTypeId } : null,
+  },
+  assets: [],
+  items: [],
+});
+
+describe("web spotlight transfomers", () => {
+  it("correctly transforms disabled web spotlight", () => {
+    const environmentModel = createEnvironmentModel(false, null);
+    const expectedOutput = { enabled: false, root_type: null };
+
+    const result = transformWebSpotlightModel(environmentModel);
+
+    expect(result).toEqual(expectedOutput);
+  });
+
+  it("correctly transforms enabled web spotlight with existing root type", () => {
+    const environmentModel = createEnvironmentModel(true, "typeId1");
+    const expectedOutput = { enabled: true, root_type: { codename: "type_codename_1" } };
+
+    const result = transformWebSpotlightModel(environmentModel);
+
+    expect(result).toEqual(expectedOutput);
+  });
+
+  it("correctly transforms enabled web spotlight with non-existing root type", () => {
```

Comment:
I would rather test when the web spotlight is deactivated but root_type isn't.. This one can happen, when you deactivate the web_spotlight :)

Moreover I am not sure, whether you can have enabled web_spotlight without any type